src/Application/SqlScripts/SqlTraits.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Ravenfire\Magpie\Application\SqlScripts;
+
+/**
+ * Traits used by the sql scripts
+ */
+trait SqlTraits
+{
+    /**
+     * Limits the width of table columns to 12 characters.
+     *
+     * @param $results
+     * @param $db_columns
+     * @return array
+     */
+    public function setStrLen($results, $db_columns)
+    {
+        foreach ($results[0] as $result => $data) {
+            $db_columns[] = $result;
+        }
+
+        foreach ($results as $result) {
+            $row = [];
+            foreach ($db_columns as $db_column) {
+                if (strlen($result->$db_column) > 12) {
+                    $result->$db_column = substr($result->$db_column, 0, 12);
```

Comment:
I **think** you could simplify this by just skipping setting it back to `$result`:

`$row[] = substr($result->$db_column, 0, 12);` But you'd want to test that.

Reply:
The goal was to create an array of arrays. Each array inside the array is set as a row. If `row` isn't set back to empty doesn't it just continue to grow in length. How  does 'setRow()' know where the first row ends and the next one begins?

Reply:
Your right. You still need to set `$row`. I'm saying you don't need to set `$result->$db_column` I don't think.

Reply:
Oh, I see what you're doing here. Yeah, it's fine. I'm not sure why it looked wonky to my eye, but I guess I was skipping over the if statement. You can disregard this comment.

I guess this is how I would have done it:

```php
        foreach ($results as $result) {
            $row = [];
            foreach ($db_columns as $db_column) {
                $row[] = (strlen($result->$db_column) > 12)
                     ? substr($result->$db_column, 0, 12)
                     : $result->$db_column
                }
            }
            $rows[] = $row;
        }
```

Though that's not really any different, and may not be as readable. Just sharing a thought.

You can ignore this comment chain :)